Game2048Player/src/test/java/gameplay/KeyEventGamePlayTest.java

```diff
@@ -0,0 +1,396 @@
+                new EmptyTile(new Posn(0,0)),
+                new Tile(2, new Posn(0,1)),
+                new EmptyTile(new Posn(0,2)),
+                new EmptyTile(new Posn(0,3))};
+        rowUpEvent011 = new Square[]{
+                new EmptyTile(new Posn(1,0)),
+                new EmptyTile(new Posn(1,1)),
+                new EmptyTile(new Posn(1,2)),
+                new EmptyTile(new Posn(1,3))};
+        rowUpEvent012 = new Square[]{
+                new EmptyTile(new Posn(2,0)),
+                new EmptyTile(new Posn(2,1)),
+                new EmptyTile(new Posn(2,2)),
+                new EmptyTile(new Posn(2,3))};
+        rowUpEvent013 = new Square[]{
+                new EmptyTile(new Posn(3,0)),
+                new EmptyTile(new Posn(3,1)),
+                new EmptyTile(new Posn(3,2)),
+                new EmptyTile(new Posn(3,3))};
+
+        gUpEvent01 = new Grid2048(new Square[][] {rowUpEvent010, rowUpEvent011, rowUpEvent012, rowUpEvent013});
+
+
+        rowEmpty0 = new Square[]{
+                new EmptyTile(new Posn(0,0)),
+                new EmptyTile(new Posn(0,1)),
+                new EmptyTile(new Posn(0,2)),
+                new EmptyTile(new Posn(0,3))};
+        rowEmpty1 = new Square[]{
+                new EmptyTile(new Posn(1,0)),
+                new EmptyTile(new Posn(1,1)),
+                new EmptyTile(new Posn(1,2)),
+                new EmptyTile(new Posn(1,3))};
+        rowEmpty2 = new Square[]{
+                new EmptyTile(new Posn(2,0)),
+                new EmptyTile(new Posn(2,1)),
+                new EmptyTile(new Posn(2,2)),
+                new EmptyTile(new Posn(2,3))};
+        rowEmpty3 = new Square[]{
+                new EmptyTile(new Posn(3,0)),
+                new EmptyTile(new Posn(3,1)),
+                new EmptyTile(new Posn(3,2)),
+                new EmptyTile(new Posn(3,3))};
+
+        g0 = new Grid2048(new Square[][] {rowEmpty0, rowEmpty1, rowEmpty2, rowEmpty3});
+
+        row0 = new Square[]{
+                new Tile(2, new Posn(0,0)),
+                new Tile(2, new Posn(0,1)),
+                new Tile(2, new Posn(0,2)),
+                new Tile(2, new Posn(0,3))};
+        row1 = new Square[]{
+                new Tile(2, new Posn(1,0)),
+                new EmptyTile(new Posn(1,1)),
+                new EmptyTile(new Posn(1,2)),
+                new Tile(2, new Posn(1,3))};
+        row2 = new Square[]{
+                new Tile(2, new Posn(2,0)),
+                new EmptyTile(new Posn(2,1)),
+                new EmptyTile(new Posn(2,2)),
+                new Tile(2, new Posn(2,3))};
+        row3 = new Square[]{
+                new Tile(2, new Posn(3,0)),
+                new Tile(2, new Posn(3,1)),
+                new Tile(2, new Posn(3,2)),
+                new Tile(2, new Posn(3,3))};
+
+        g1 = new Grid2048(new Square[][] {row0, row1, row2, row3});
+
+    }
+
+    @Test
+    public void testNoTilesMovedKeyHandler () {
```

Comment:
Do you like `itDoesntMoveTilesWhenMoveCannotBeMade` or something like that?

Reply:
I love it!!!